src/core/structures/PDFRectangle.ts

```diff
@@ -0,0 +1,46 @@
+import { PDFArray, PDFNumber } from 'src/core';
+
+class PDFRectangle {
+  static fromArray(array: PDFArray) {
+    return new PDFRectangle(array);
+  }
+
+  readonly array: PDFArray;
+
+  protected constructor(array: PDFArray) {
+    this.array = array;
+  }
+
+  getNormalized(): PDFRectangle {
+    const normalized = this.array.clone(this.array.context);
+
+    if (normalized.lookup(0, PDFNumber) > normalized.lookup(2, PDFNumber)) {
+      normalized.set(0, this.array.lookup(2, PDFNumber));
+      normalized.set(2, this.array.lookup(0, PDFNumber));
+    }
+
+    if (normalized.lookup(1, PDFNumber) > normalized.lookup(3, PDFNumber)) {
+      normalized.set(1, this.array.lookup(3, PDFNumber));
+      normalized.set(3, this.array.lookup(1, PDFNumber));
+    }
+    return PDFRectangle.fromArray(normalized);
+  }
```

Comment:
1. When do you see this operation being useful? I'm not sure I understand its utility.
2. The `normalized` array being created here is an orphan. It is not attached to its context in any way. So if somebody modifies it, the changes will not be reflected in the serialized document (unless they attach it themselves). I'm not sure what the best way to resolve this is. It _may_ not be an issue either. It really depends on how this method is intended to be used (which leads back to my first question).

Reply:
I could imagine a user wanting the AcroFormsAPI to let them create fields and manipulate their dimensions. If you think this is out of scope I can cut it

Reply:
@cshenks I agree that this will need to be possible. My concern here is that the newly created `PDFArray` is not actually attached to the context. I think the way to do this would be to define a method like the following:
```js
PDFAnnotation.setRectangle({
  x: number,
  y: number,
  width: number,
  height: number,
})
```

Reply:
@Hopding Broke this out into #320